pkg/nri/config.go

```diff
@@ -42,7 +42,7 @@ type Config struct {
 // DefaultConfig returns the default configuration.
 func DefaultConfig() *Config {
 	return &Config{
-		Disable:          true,
```

Comment:
I think we can suggest user to use `disabled_plugins` to disable NRI. Anyway, our code should handle if NRI plugin is empty.

Reply:
I haven't though of that... But what about configuration migration ? I think with the current plumbing it would not be possible to automatically migrate a 1.7 configuration (which has NRI disabled in the plugin's own configuration)  to a 2.0 configuration which then would have NRI disabled by the disabling the plugin itself in the configuration's global section, because IIUC the plugin cannot alter/does not have access to it.

Reply:
Yeah. It's hard to handle migration. However, we should remove the `disable` in the future since it's not good patten in plugin design. (I'm still thinking that 2.0 is good slot to remove it)

Reply:
> Yeah. It's hard to handle migration. However, we should remove the `disable` in the future since it's not good patten in plugin design. (I'm still thinking that 2.0 is good slot to remove it)

Is it completely out of the question to update the current configuration migration infra so that, in addition to changing its own configuration, a plugin could ask to be added to the list of disabled plugins (for instance via an extra boolean returned from ConfigMigration() ) ?

Reply:
We can document it, but agreed that we should keep the config for now. There are many cases where a plugin will skip itself based on the configuration.

Reply:
agree to keep it for now new 2.0 users can enable the disable for an extra security check and we can make explicit mention in the release notes.. possibly noting to admins what they should to do to verify NRI plugins